### DISCUSSION TITLE
Search for docs: onsite / via google

### DISCUSSION THREAD

**Discussion by hpvd** [terminusdb/terminusdb#2062] [Ideas]
would be great to have the possibility to onsite search within docs (like possible on your main site)
Otherwise you always have to rely on google,
which for this task today does not give great results because they seem to index also navigation menu on doc pages... (possibly because of code structure)

example:  search for kubernetes and you will receive to many result pages:
`site:https://terminusdb.com/docs/ kubernetes`

**Comment by hpvd** [comment]
for the google route:

maybe it's not the code structure alone, but a sitemap related issue.

One probably could help to make it work for google by fixing this:
- general structure: the main sitemap points to 5 sub-sitemaps (ok)
- there is a redirection needed to reach it from standard url https://terminusdb.com/sitemap.xml to https://terminusdb.com/sitemap_index.xml (ok, but even better without)
- main sitemap shows a `last modified date` of sub-sitemap of doc of `2023-15-06`, but when looking into this sub-sitemap there is shown `2023-09-27T` (issue: google may stop at main sitemap because of having already visited the sub-sitemap after 2023-15-06)
- in sub-sitemap, all pages have the same change date `<lastmod>2023-09-27T06:15:23.272Z</lastmod>` (issue: google may think the sitemap is buggy and will not take it into account)

all this may lead to the index/result issue described in the example above (kubernetes from nav is identified as content on every page)

=> Of course a dedicated onsite search for docs would be far more handy :-)

**Comment by alexander-mart** [comment]
@hpvd I add you idea to the list of proposals about documentation: #2157
